nearswap/tests/fee_simulation.rs

```diff
@@ -0,0 +1,234 @@
+    let mut before_swap_token = view!(
+        nearswap.get_deposit_token(alice.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    assert_close(before_swap_token, to_yocto("0"), 0);
+
+    let expected_receive = view!(
+            nearswap.price_near_to_token_in(dai(), to_yocto_str("5"))
+        ).unwrap_json::<U128>();
+
+    let mut pool_before = view!(nearswap.pool_info(&dai())).unwrap_json::<PoolInfo>();
+
+    call!(
+        alice,
+        nearswap.swap_near_to_token_exact_in(to_yocto_str("5"), dai(), expected_receive),
+        deposit = 1
+    ).assert_success();
+
+    let mut after_swap_token = view!(
+            nearswap.get_deposit_token(alice.account_id, dai())
+        ).unwrap_json::<U128>();
+
+    assert_eq!(
+        to_u128(before_swap_token) + to_u128(expected_receive),
+        to_u128(after_swap_token), "Near to token swap unsuccessful");
+
+    // Check if fee is deducted - Near deposited into pool must be 0.997 * near amount
+    // Fee - 0.3%
+    let out = mock_calc_out(
+        to_yocto("5")*997/1000,
+        to_u128(pool_before.ynear), to_u128(pool_before.tokens)
+    );
+    assert_eq!(out, to_u128(after_swap_token), "Wrong amount of fee deducted");
+
+    pool_before = view!(nearswap.pool_info(&dai())).unwrap_json::<PoolInfo>();
+    let before_swap_token_lp1 = view!(
+        nearswap.get_deposit_token(lp1.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    let before_swap_near_lp1 = view!(
+        nearswap.get_deposit_near(lp1.account_id.clone())
+    ).unwrap_json::<U128>();
+    let before_swap_token_lp2 = view!(
+        nearswap.get_deposit_token(lp2.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    let before_swap_near_lp2 = view!(
+        nearswap.get_deposit_near(lp2.account_id.clone())
+    ).unwrap_json::<U128>();
+
+    // withdraw liquidity
+    call!(
+        lp1,
+        nearswap.withdraw_liquidity(dai(), lp1_shares, U128(1), U128(1))
+    ).assert_success();
+
+    let pool_after = view!(nearswap.pool_info(&dai())).unwrap_json::<PoolInfo>();
+    let after_swap_token_lp1 = view!(
+        nearswap.get_deposit_token(lp1.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    let after_swap_near_lp1 = view!(
+        nearswap.get_deposit_near(lp1.account_id.clone())
+    ).unwrap_json::<U128>();
+
+    // Check If ~90% of total shares are received by lp1
+    let tokens_received_lp1 = to_u128(after_swap_token_lp1) - to_u128(before_swap_token_lp1);
+    assert_eq!(to_u128(pool_before.tokens)*9/10, tokens_received_lp1, "Redeemed liquidity is not correct for lp1");
+
+    let near_received_lp1 = to_u128(after_swap_near_lp1) - to_u128(before_swap_near_lp1);
+    assert_eq!(to_u128(pool_before.ynear)*9/10, near_received_lp1, "Redeemed Near incorrect - lp1");
+    
+    call!(
+        lp2,
+        nearswap.withdraw_liquidity(dai(), lp2_shares, U128(1), U128(1))
+    ).assert_success();
+
+    let after_swap_token_lp2 = view!(
```

Comment:
as above, let's don't use swap in the name.

Reply:
ok